### DISCUSSION TITLE
How much do the automatic optimizations actually do?

### DISCUSSION THREAD

**Discussion by Mabi19** [nuxt/nuxt#17177] [Questions] [answered]
I have a project which essentially has two entirely separate parts - the user-facing section and the admin panel. I've been carefully tracking the download sizes for several pages in both sections, and I've noticed a couple of peculiar things:

- After adding a large dependency to the admin panel, the download size when going to the root page increased by the size of the dependency
- This does not happen when the component actually using the dependency is loaded lazily
- All layouts (and I suspect that pages at least partially) are always loaded

Therefore, I want to know more about what happens under the hood with the optimizations (especially code-splitting), so I can improve my app.

**Comment by TheAlexLichter** [comment, marked as answer]
Hey 👋🏻

Optimizations are constantly WIP to ensure Nuxt applications are as performant as possible by default.
The issue with prefetching/preloading all pages and layouts is tracked in https://github.com/nuxt/framework/issues/4475 and will be eventually fixed ☺️
See also https://github.com/nuxt/framework/pull/6210

**Reply by Mabi19** [reply]
Ah. I assumed this was some kind of intended behaviour. Good to know it's getting corrected.
